### PR TITLE
feat(train): freeze target components and support param-list in BF16Optimizer

### PR DESCRIPTION
**Pull request by curnane-lab**
## Motivation

When training DFlash with `FSDP(use_orig_params=True)`, two issues arise:

1. **Parameter reference mismatch**: Passing the unwrapped `draft_model` to `BF16Optimizer` causes it to call `draft_model.parameters()`, which returns the original parameter objects. However, `FSDP(use_orig_params=True)` manages its own parameter references via `dflash_model.named_parameters()`. The optimizer must operate on the FSDP-managed parameters to ensure gradient synchronization works correctly across ranks.

2. **Defensive freeze for target components**: While `TargetEmbeddingsAndHead.from_pretrained()` sets `requires_grad_(False)`, explicitly freezing `lm_head` and `embed_tokens` before FSDP wrapping adds a defensive layer against accidental unfreezing by future code changes.

This PR fixes both issues to improve training stability on both CUDA and NPU.

## Modifications

- **`specforge/optimizer.py`**:
  - Extend `BF16Optimizer.__init__()` to accept either an `nn.Module` or a pre-filtered `list[nn.Parameter]`
  - When a list is passed, set `self.model = None` and filter `p.requires_grad`
  - Backward compatible: existing code passing `BF16Optimizer(draft_model, ...)` still works

- **`scripts/train_dflash.py`**:
  - After `OnlineDFlashModel` creation, explicitly freeze `target_components.lm_head` and `target_components.embed_tokens` with `requires_grad = False`
  - After FSDP wrapping, collect only `draft_model.*` parameters from `dflash_model.named_parameters()`
  - Pass the filtered `draft_params` list to `BF16Optimizer` instead of the raw `draft_model`

## Related Issues

N/A (new feature)

## Accuracy Test

- [x] Not applicable — no model architecture or kernel changes; this is a training loop refactor.

## Benchmark & Profiling

- [x] Verified backward compatibility: `BF16Optimizer(model, ...)` still works for non-FSDP training.
- [x] With `use_orig_params=True`, optimizer now operates on FSDP-managed parameter references, avoiding potential gradient sync issues.

## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [x] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).

### CODE REVIEW

File: specforge/optimizer.py [hunk 1]
```diff
@@ -18,8 +18,16 @@ def __init__(
...
+        if isinstance(model_or_params, list):
+            self.model = None
+            self.model_params = [
+                p for p in model_or_params if p.requires_grad
+            ]
+        else:
+            self.model = model_or_params
+            self.model_params = [
+                p for p in model_or_params.parameters() if p.requires_grad
+            ]
```
[the remark this answers is not in the document]
Comment (curnane-lab) [reply specforge/optimizer.py]: Thanks for the catch! Fixed in the latest commit (0201117). The implementation now consistently uses model (the parameter name from the signature) and checks isinstance(model, list) to support both nn.Module and pre-filtered parameter lists. Backward compatibility is preserved.
